samples/drivers/dac/src/main.c

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2020 Libre Solar Technologies GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <sys/printk.h>
+#include <drivers/dac.h>
+
+#define DAC_DEVICE_NAME		DT_DAC_1_NAME
+#define DAC_RESOLUTION		12
+#define DAC_1ST_CHANNEL_ID	1
+
+static const struct dac_channel_cfg dac_ch1_cfg = {
+	.channel_id  = DAC_1ST_CHANNEL_ID,
+	.buffered = false,
+};
+
+void main(void)
+{
+	struct device *dac_dev = device_get_binding(DAC_DEVICE_NAME);
+
+	if (!dac_dev) {
+		printk("Cannot get DAC device\n");
+		return;
+	}
+
+	int ret = dac_channel_setup(dac_dev, &dac_ch1_cfg);
+
+	if (ret != 0) {
+		printk("Setting up of channel 1 failed with code %d\n", ret);
+		return;
+	}
+
+	/* set half of supply/reference voltage */
+	dac_write(dac_dev, DAC_1ST_CHANNEL_ID, 2048, 12);
```

Comment:
Yep, no problem. What about generating a saw-tooth signal in a for loop with an interval of one minute so that you can still observe it with a multi-meter?

Reply:
Indeed, maybe something a 1 sec period would be ok. It can even be seen using a led.

We can also think of something bit more elaborated and feedback the output to a gpio and verify input level. This way it could be tested with only a simple wire (could be left for future work).

Reply:
1 sec period was too slow to be seen with a multimeter, so I took 4 sec (counting from 0 to 4095 with 1 msec sleep).